SSA/Projects/InstCombine/HackersDelight.lean

```diff
@@ -459,6 +459,150 @@ theorem div_overflow_iff_neq_and_RightShift_lt {x y : BitVec 64} {y : BitVec 32}
   try alive_auto
   all_goals sorry
 
+def signedDifferenceOrZero (x y : BitVec w) : BitVec w :=
+    if x ≥ₛ y then x - y else 0#w
+
+def unsignedDifferenceOrZero (x y : BitVec w) : BitVec w :=
+    if x ≥ᵤ y then x - y else 0#w
+
+def unsignedMaxBitVec (x y : BitVec w) : BitVec w :=
+  if x ≥ᵤ y then x else y
+
+def unsignedMinBitVec (x y : BitVec w) : BitVec w :=
+  if x ≤ᵤ y then x else y
+
+def signedMaxBitVec (x y : BitVec w) : BitVec w :=
```

Comment:
It's up to you, but maybe I would name this definition something like "BitVec.signedMax".